### PR TITLE
Add a helper that returns all platforms know to supermarket

### PR DESCRIPTION
**Pull request by tas50**
Creates a single place for us to add new platforms. This will be used by the "any" concept in a bit

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

**Comment by nellshamrell** [comment]
👍

**Comment by tas50** [comment]
I should mention that I have no local supermarket to test this on (and I'd love to get that setup sometime)

File: src/supermarket/app/helpers/supported_platforms_helper.rb [hunk 1]
```diff
@@ -1,4 +1,39 @@
...
+      oracle
+      redhat
+      ubuntu
+      scientific
+      smartos
+      solaris
+      suse
+      windows
+      zlinux)
+  end
```
Comment (robbkidd) [inline src/supermarket/app/helpers/supported_platforms_helper.rb]: I have a not-fully-formed feeling that this canonical list of known platform names should live deeper, like as a class constant in the `SupportedPlatforms` model. That could still be bubbled up to views through this helper method.

Comment (robbkidd) [inline src/supermarket/app/helpers/supported_platforms_helper.rb]: I _do_ agree that we should have the canonical list _somewhere_ to use consistently.

**Comment by tas50** [comment]
After realizing we were missing a lot of platforms this is going to take some rework. See 5a58c1ea23012e3a7f54632c7b2174e7fafb8cdc for the large list

**Pull request closed without merging**